### PR TITLE
Import new model tiiuae falcon 7b

### PR DESCRIPTION
**Pull request by MallikharjunaThota123**
Model :- tiiuae/falcon-7b

FT and evaluation link:- https://ml.azure.com/runs/mango_rod_y5rhtfkw2g?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourcegroups/shared-finetuning-rg/workspaces/v-mathota&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Save blob run link:- https://ml.azure.com/experiments/id/01281a62-36ec-46dd-83a0-408c1269f22b/runs/willing_airport_z6bxq15mbv?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/v-mathota&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Default deployement endpoint response:
Default Deploy for Falcon 7b

Batch Deployment :- https://ml.azure.com/experiments/id/e173bb89-668f-4724-86b2-703efff7017d/runs/batchjob-d8599016-53cc-4315-a9b1-edb65861ab78?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/v-mathota&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Falcon 7b batch deployment version 13

conda.yaml

```
channels:
- conda-forge
dependencies:
- python=3.8.17
- pip<=23.2.1
- pip:
  - mlflow==2.3.1
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.3.1
  - azureml-core==1.52.0
  - azureml-mlflow==1.52.0
  - azureml-metrics==0.0.24
  - scikit-learn==1.1.0
  - cryptography==41.0.2
  - python-dateutil==2.8.2
  - datasets==2.14.4
  - soundfile==0.12.1
  - librosa==0.10.0.post2
  - diffusers==0.19.3
  - sentencepiece==0.1.99
  - transformers==4.31.0
  - torch==2.0.0
  - accelerate==0.21.0
  - Pillow==9.4.0
  - einops
  - azureml-evaluate-mlflow==0.0.24
- pycocotools=2.0.4
name: mlflow-env
```

MLmodel:-

```
flavors:
  hftransformersv2:
    code: null
    config_hf_load_kwargs:
      trust_remote_code: true
    hf_config_class: AutoConfig
    hf_pretrained_class: AutoModelForCausalLM
    hf_tokenizer_class: AutoTokenizer
    huggingface_id: tiiuae/falcon-7b
    model_data: data
    model_hf_load_args:
      trust_remote_code: true
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    tokenizer_hf_load_kwargs:
      trust_remote_code: true
    transformers_version: 4.31.0
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.17
mlflow_version: 2.3.1
model_uuid: bf84c6fac15b483fa54bfc1edbe5c17e
utc_time_created: '2023-09-25 11:46:10.329850'
```

### CODE REVIEW

**Comment by novaturient95** [comment]
LGTM

File: assets/models/system/tiiuae-falcon-7b/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  finetune_compute_allow_list:
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  model_specific_defaults:
+    apply_lora: 'true'
+    precision: '4'
+version: 2
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/spec.yaml:46]: set version to 1

File: assets/models/system/tiiuae-falcon-7b/description.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.sh</a>
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+### Model Evaluation Sample
+
+Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
+|--|--|--|--|--|
+Text generation | Text generation | <a href="https://huggingface.co/datasets/cnn_dailymail" target="_blank"> cnn_dailymail </a> | <a href="https://aka.ms/azureml-eval-sdk-text-generation/" target="_blank">evaluate-model-text-generation.ipynb</a> | <a href="https://aka.ms/azureml-eval-cli-text-generation/" target="_blank">evaluate-model-text-generation.yml</a>
+
```
Comment (skasturi) [inline assets/models/system/tiiuae-falcon-7b/description.md:105]: Could you please add FT samples as well?

File: assets/models/system/tiiuae-falcon-7b/model.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+path:
+  container_name: models
+  container_path: huggingface/tiiuae-falcon-7b/1696411650/mlflow_model_folder
+  storage_name: automlcesdkdataresources
+  type: azureblob
+publish:
+  description: description.md
+  type: mlflow_model
```
Comment (skasturi) [inline assets/models/system/tiiuae-falcon-7b/model.yaml:8]: Add new line at the end of the file.

**Comment by novaturient95** [comment]
Please wait for FT to complete

File: assets/models/system/tiiuae-falcon-7b/description.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+Falcon-7B is made available under the Apache 2.0 license.
+
+# Inference samples
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.sh</a>
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+### Model Evaluation Sample
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/description.md:100]: add above inference samples. Use Header 1

File: assets/models/system/tiiuae-falcon-7b/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+properties:
+  SHA: f7796529e36b2d49094450fb038cc7c4c86afa44
+  datasets: tiiuae/falcon-refinedweb
+  languages: en
+  inference-min-sku-spec: 8|0|64|128
+  inference-recommended-sku: Standard_NC6s_v3
+  evaluation-min-sku-spec: 16|0|64|128
+  evaluation-recommended-sku: Standard_NC24s_v3
+  finetune-recommended-sku: Standard_NC24s_v3
+  finetuning-tasks: text-classification, question-answering
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/spec.yaml:13]: remove question-answering if FT has failed for it

File: assets/models/system/tiiuae-falcon-7b/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+tags:
+  Featured: ''
+  Preview: ''
+  license: apache-2.0
+  task: text-generation
+  author: tiiuae
+  inference_compute_allow_list:
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/spec.yaml:21]: evaluation compute allowlist missing

File: assets/models/system/tiiuae-falcon-7b/spec.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: tiiuae-falcon-7b
+path: ./
+properties:
+  SHA: f7796529e36b2d49094450fb038cc7c4c86afa44
+  datasets: tiiuae/falcon-refinedweb
+  languages: en
+  inference-min-sku-spec: 8|0|64|128
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/spec.yaml:8]: remove min-sku-spec for all scenarios

File: assets/models/system/tiiuae-falcon-7b/description.md [hunk 8]
```diff
@@ -103,6 +103,13 @@ Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
 |--|--|--|--|--|
 Text generation | Text generation | <a href="https://huggingface.co/datasets/cnn_dailymail" target="_blank"> cnn_dailymail </a> | <a href="https://aka.ms/azureml-eval-sdk-text-generation/" target="_blank">evaluate-model-text-generation.ipynb</a> | <a href="https://aka.ms/azureml-eval-cli-text-generation/" target="_blank">evaluate-model-text-generation.yml</a>
 
+# Finetuning samples
+
+Task|Use case|Dataset|Python sample (Notebook)|CLI with YAML
+
+|--|--|--|--|--|
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/description.md:110]: Please structure properly

Comment (novaturient95) [reply assets/models/system/tiiuae-falcon-7b/description.md:110]: image

File: assets/models/system/tiiuae-falcon-7b/description.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+#### License
+
+Falcon-7B is made available under the Apache 2.0 license.
+
+# Inference samples
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation-dolly" target="_blank">text-generation-online-endpoint-dolly.sh</a>
```
Comment (novaturient95) [inline assets/models/system/tiiuae-falcon-7b/description.md:97]:
```suggestion
Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation-dolly" target="_blank">text-generation-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation-dolly" target="_blank">text-generation-online-endpoint.sh</a>
```

File: assets/models/system/tiiuae-falcon-7b/description.md (hunk 8 above)
Comment (novaturient95) [reply assets/models/system/tiiuae-falcon-7b/description.md]: Done

File: assets/models/system/tiiuae-falcon-7b/description.md (hunk 2 above)
Comment (novaturient95) [reply assets/models/system/tiiuae-falcon-7b/description.md:106]: Done

**Review by novaturient95 — APPROVED** [review]

File: assets/models/system/tiiuae-falcon-7b/spec.yaml (hunk 6 above)
Comment (novaturient95) [reply assets/models/system/tiiuae-falcon-7b/spec.yaml:20]: Done

File: assets/models/system/tiiuae-falcon-7b/model.yaml (hunk 3 above)
Comment (novaturient95) [reply assets/models/system/tiiuae-falcon-7b/model.yaml]: Done

**Pull request merged**